src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.js

```diff
@@ -0,0 +1,205 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { ILLEGAL_CHARACTERS } from '../../constants';
+import { getIndices } from '../../lib/get_indices';
+import { isIndexPatternQueryValid } from '../../lib/is_index_pattern_query_valid';
+import { getWhitelistedIndices } from '../../lib/get_whitelisted_indices';
+import { LoadingIndices } from './components/loading_indices';
+import { StatusMessage } from './components/status_message';
+import { IndicesList } from './components/indices_list';
+
+import {
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiSpacer,
+  EuiButton,
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+} from '@elastic/eui';
+import { canAppendWildcard } from '../../lib/can_append_wildcard';
+import { createReasonableWait } from '../../lib/create_reasonable_wait';
+
+export class StepIndexPattern extends Component {
+  static propTypes = {
+    initialIndices: PropTypes.array.isRequired,
+    isIncludingSystemIndices: PropTypes.bool.isRequired,
+    esService: PropTypes.object.isRequired,
+    goToNextStep: PropTypes.func.isRequired,
+    initialQuery: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      matchingIndices: [],
+      isLoadingIndices: false,
+      query: props.initialQuery || '',
```

Comment:
We use `defaultProps` to pass in an empty string instead of applying a default here.

Reply:
Sounds good!